frontend/pages/api/v1/all_locked_accounts.ts

```diff
@@ -0,0 +1,118 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { PythBalance } from '@pythnetwork/staking/app/pythBalance'
+import BN from 'bn.js'
+import { STAKING_ADDRESS } from '@pythnetwork/staking/app/constants'
+import { Connection, Keypair } from '@solana/web3.js'
+import { bs58 } from '@coral-xyz/anchor/dist/cjs/utils/bytes'
+import { Program, AnchorProvider, IdlAccounts } from '@coral-xyz/anchor'
+import NodeWallet from '@coral-xyz/anchor/dist/cjs/nodewallet'
+import { Staking } from '@pythnetwork/staking/lib/target/types/staking'
+import idl from '@pythnetwork/staking/target/idl/staking.json'
+import { splTokenProgram } from '@coral-xyz/spl-token'
+import { TOKEN_PROGRAM_ID } from '@solana/spl-token'
+import {
+  getCustodyAccountAddress,
+  getMetadataAccountAddress,
+} from './locked_accounts'
+
+const ONE_YEAR = new BN(3600 * 24 * 365)
+
+const connection = new Connection(process.env.BACKEND_ENDPOINT!)
+const provider = new AnchorProvider(
+  connection,
+  new NodeWallet(new Keypair()),
+  {}
+)
+const stakingProgram = new Program<Staking>(
+  idl as Staking,
+  STAKING_ADDRESS,
+  provider
+)
+const tokenProgram = splTokenProgram({
+  programId: TOKEN_PROGRAM_ID,
+  provider: provider as any,
+})
+
+export default async function handlerAllLockedAccounts(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const allStakeAccounts = await getAllStakeAccounts(connection)
+
+  const allMetadataAccountAddresses = allStakeAccounts.map((account) =>
+    getMetadataAccountAddress(account)
+  )
+  const allCustodyAccountAddresses = allStakeAccounts.map((account) =>
+    getCustodyAccountAddress(account)
+  )
+
+  const allMetadataAccounts =
+    await stakingProgram.account.stakeAccountMetadataV2.fetchMultiple(
+      allMetadataAccountAddresses
+    )
+  const allCustodyAccounts = await tokenProgram.account.account.fetchMultiple(
+    allCustodyAccountAddresses
+  )
+
+  const lockedCustodyAccounts = allCustodyAccounts
+    .map((data, index) => {
+      return { pubkey: allCustodyAccountAddresses[index], data }
+    })
+    .filter((account, index) => {
+      const metadataAccountData = allMetadataAccounts[index]
+      return (
+        metadataAccountData &&
+        account.data &&
+        hasStandardLockup(metadataAccountData)
+      )
+    })
+    .sort((a, b) => {
+      return a.data!.amount.lte(b.data!.amount) ? 1 : -1
+    }) // ! is safe because of the filter above
+
+  const totalLockedAmount = new PythBalance(
+    lockedCustodyAccounts.reduce((total, account) => {
+      return total.add(account.data ? new BN(account.data.amount) : new BN(0))
+    }, new BN(0))
+  )
+
+  res.status(200).json({
+    totalLockedAmount: totalLockedAmount.toString(),
```

Comment:
the PythBalances are printing like "1,000,000" which is going to be annoying for programmatic users. (I think you do want the balance to be a string to avoid precision issues, but without the commas)

Reply:
Will fix later